Client/Frontend/Menu/Items/MenuItemView.swift

```diff
@@ -0,0 +1,106 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+
+/**
+ * creates a view that consists of an image and a title.
+ * the image view is displayed at the center top of the
+ **/
+public class MenuItemView: UIControl {
+
+    var padding: CGFloat = 5.0
+
+    private(set) public var imageView: UIImageView
+    private(set) public var titleLabel: UILabel
+
+    private var previousLocation: CGPoint?
+    public init() {
+        imageView = UIImageView()
+        titleLabel = UILabel()
+
+        super.init(frame: CGRectZero)
+
+        titleLabel.lineBreakMode = NSLineBreakMode.ByWordWrapping
+        titleLabel.numberOfLines = 2
+        titleLabel.textAlignment = NSTextAlignment.Center
+
+        self.addSubview(imageView)
+
+        self.addSubview(titleLabel)
+
+        imageView.snp_makeConstraints { make in
+            make.centerX.equalTo(self)
+            make.bottom.equalTo(self.snp_centerYWithinMargins)
+        }
+
+        titleLabel.snp_makeConstraints { make in
+            make.top.equalTo(imageView.snp_bottom).offset(padding)
+            make.centerX.equalTo(self)
+            make.bottom.equalTo(self).offset(-padding)
+            make.width.lessThanOrEqualTo(self.snp_width).dividedBy(1.5)
+        }
+    }
+
+    required public init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    public override func layoutSubviews() {
+        super.layoutSubviews()
+        titleLabel.preferredMaxLayoutWidth = self.bounds.size.width / 1.5
+    }
+
+    func prepareForReuse() {
+        imageView.image = nil
+        titleLabel.text = nil
+
+        guard let recognisers = gestureRecognizers else { return }
+        for recogniser in recognisers {
+            if recogniser is UILongPressGestureRecognizer {
+                self.removeGestureRecognizer(recogniser)
+            }
+        }
+
+        self.snp_removeConstraints()
+    }
+
+    public func setTitle(title: String) {
+        titleLabel.text = title
+    }
+
+    public func setImage(image: UIImage) {
```

Comment:
If you have these accessor methods for `imageView` and `textLabel` you probably don't need to expose those views publicly at all and keep them private.